client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -1183,23 +1166,21 @@ public void onSuccess(ByteBuffer response) {
           @Override
           public void onFailure(Throwable e) {
             StatusCode cause = getPushDataFailCause(e.getMessage());
+
             if (pushState.exception.get() != null) {
               return;
             }
             if (remainReviveTimes <= 0) {
-              callback.onFailure(
-                  new Exception(
-                      cause
-                          + "! Push data to master worker of "
-                          + hostPort
-                          + " failed: "
-                          + e.getMessage(),
-                      e));
+              if (e instanceof CelebornIOException) {
+                callback.onFailure(e);
+              } else {
+                callback.onFailure(new CelebornIOException(cause, e));
+              }
               return;
             }
             logger.error(
                 "Push merged data to {} failed for shuffle {} map {} attempt {} partition {} groupedBatch {} batch {}.",
-                hostPort,
+                addressPair,
```

Comment:
Better to print ```remainReviveTimes``` here, and should log in else branch below that mapper has ended

Reply:
DOne